ecl/hql/hqlmanifest.cpp

```diff
@@ -198,6 +198,31 @@ void ResourceManifest::loadResource(const char *filepath, MemoryBuffer &content)
     read(fio, 0, (size32_t) f->size(), content);
 }
 
+void queryJfrog(const char *localpath, IPropertyTree &item)
+{
+    const char *repo = item.queryProp("@repository");
+    if(repo)
+    {
+        StringBuffer filename(localpath);
+        getFileNameOnly(filename, false);
+        StringBuffer filepath;
+        StringBuffer machine;
+        if (repo[0] == '/' && repo[1] == '/')
+            splitUNCFilename(repo, &machine, &filepath, nullptr, nullptr);
+        else 
+            splitFilename(repo, nullptr, &filepath, nullptr, nullptr);
+        StringBuffer jfrog_cmd("export CI=true; jf rt dl --flat --quiet ");
+        if (!machine.isEmpty())
+            jfrog_cmd.appendf("--url=http:%s/artifactory ", machine.str());
+        StringBuffer secretValue;
+        const char *user = item.queryProp("@user");
+        if (user && getSecretValue(secretValue.clear(), "jfrog", user, "key", true))
```

Comment:
Is the user the correct key to use here?  I suspect the repository might be a better bet.  Otherwise there is potential for exfiltration of secrets.

Reply:
Changed the key to the repository name.